c/src/include/libsbp/internal/v4/string/double_null_terminated.h

```diff
@@ -10,6 +10,56 @@
  * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
  */
 
+/**
+ * @file double_null_terminated.h
+ *
+ * Handles encoding and decoding of double NULL terminated strings.
+ *
+ * In SBP double NULL terminated strings are rather similar to multipart strings
+ * except they have an extra NULL terminator at the end of a sequence. A double
+ * NULL terminated string is not a single C style string, rather it is a
+ * collection of substrings (or sections) each separated on the wire with a NULL
+ * terminator with the end of sequence being marked by an extra sequence.
+ *
+ * For example, a double NULL terminated string might contain 3 sections -
+ * "one", "two", and "three". On the wire this string would be encoded as
+ *
+ * one\0two\0three\0\0
+ *
+ * for a total of 15 bytes.
+ *
+ * A double NULL terminated string might contain no sections in which case on
+ * the wire it would consists of just two NULL terminators with no printable
+ * text.
+ *
+ * Since C strings are NULL terminated it isn't possible for a consumer of
+ * libsbp to interact directly with the wire encoding of a double NULL
+ * terminated string without performing some sort of tokenisation on the input
+ * or carefully constructing the output.
+ *
+ * The functions in this file handle all aspects of interacting with a double
+ * NULL terminated string for a consumer of libsbp, including all encoding and
+ * decoding considerations.
+ *
+ * When processing an incoming double NULL terminated string individual sections
+ * can be retrieved from the double NULL terminated string object with
+ * #sbp_double_null_terminated_string_get_section. The returned string will
+ * corrospond to the requested section number and will always be represented in
+ * C format. For example when operating on the string from the first example
+ * above, calling #sbp_double_null_terminated_string_get_section(&string,
```

Comment:
Not sure how this will render in doxygen, might just want to quote the whole thing with backticks